langchain/retrievers/tfidf.py

```diff
@@ -21,10 +21,18 @@ class Config:
         arbitrary_types_allowed = True
 
     @classmethod
-    def from_texts(cls, texts: List[str], **kwargs: Any) -> "TFIDFRetriever":
+    def from_texts(
+        cls,
+        texts: List[str],
+        tfidf_params: Optional[Dict[str, Any]] = None,
+        **kwargs: Any
+    ) -> "TFIDFRetriever":
         from sklearn.feature_extraction.text import TfidfVectorizer
 
-        vectorizer = TfidfVectorizer()
+        if tfidf_params is None:
+            vectorizer = TfidfVectorizer()
+        else:
+            vectorizer = TfidfVectorizer(**tfidf_params)
```

Comment:
```python
tfidf_params = tfidf_params or {}
vectorizer = TfidfVectorizer(**tfidf_params)
```

Reply:
@dev2049 
Thank you for review comment.
It is smart way, I just updated!